src/psp.ts

```diff
@@ -0,0 +1,502 @@
+                name: 'setter',
+                type: 'address',
+              },
+              {
+                internalType: 'uint64',
+                name: 'expirationTimestamp',
+                type: 'uint64',
+              },
+            ],
+            stateMutability: 'view',
+            type: 'function',
+          },
+          {
+            inputs: [
+              {
+                internalType: 'bytes[]',
+                name: 'data',
+                type: 'bytes[]',
+              },
+            ],
+            name: 'multicall',
+            outputs: [
+              {
+                internalType: 'bytes[]',
+                name: 'results',
+                type: 'bytes[]',
+              },
+            ],
+            stateMutability: 'nonpayable',
+            type: 'function',
+          },
+        ];
+
+        const subscriptionIds: string[] = [];
+        for (const { address, startIndex, endIndex } of chain.allocators) {
+          const allocator = new ethers.Contract(address, allocatorAbi, provider);
+          const staticCalldatas = [];
+          for (let i = startIndex; i <= endIndex; i++) {
+            staticCalldatas.push(
+              allocator.interface.encodeFunctionData('airnodeToSlotIndexToSlot', [airnodeAddress, i])
+            );
+          }
+          // TODO: retryGo?
+          const resultDatas = await allocator.callStatic.multicall(staticCalldatas);
+          resultDatas.forEach((resultData: ethers.utils.BytesLike) => {
+            const result = allocator.interface.decodeFunctionResult('airnodeToSlotIndexToSlot', resultData);
+            if (result.subscriptionId && result.expirationTimestamp.gt(nowInSeconds())) {
+              subscriptionIds.push(result.subscriptionId);
+            }
+          });
+        }
+
+        // **************************************************************************
+        // Process each psp subscription in serial to keep nonces in order
+        // **************************************************************************
+        for (const subscriptionId of subscriptionIds) {
+          const subscriptionIdLogOptions = {
+            ...providerLogOptions,
+            additional: {
+              subscriptionId,
+            },
+          };
+
+          // **************************************************************************
+          // Fetch subscription details
+          // **************************************************************************
+          node.logger.debug('Fetching subscription details...', subscriptionIdLogOptions);
+
+          // On-chain details: AirnodeProtocol.subscriptions(id)
+          // Off-chain details: triggers.psp
+          const subscription = triggers.psp.find((s) => {
+            // TODO: Should the config contain subscriptionId field
+            // or should we derive it using the other fields to verify
+            // that values in config are correct?
```

Comment:
The config file will include the subscription/template ID, the node use these IDs to get the respective details, then recalculate the ID with the details to verify it

Reply:
I still need to add the check for validating the subscriptionId matches the subscription details